### PR TITLE
Fix documentation builds

### PR DESCRIPTION
**Pull request by Zeitsperre**
### Pull Request Checklist:
- [ ] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fixes #xyz
- [ ] (If applicable) Documentation has been added / updated (for bug fixes / features)
- [x] This PR does not seem to break the templates.
- [x] HISTORY.rst has been updated (with summary of main changes)
  - [x] Link to issue (:issue:`number`) and pull request (:pull:`number`) has been added

### What kind of change does this PR introduce?

* Adds some Makefile recipes for the documentation build steps
* Removes the llvmlite reinstallation step that was needed a few years ago to deal with some weirdness with ReadTheDocs
* Applies a consistent `package <comparison><version>` schema to environment files

### Does this PR introduce a breaking change?

No.

### CODE REVIEW

**Comment by RondeauG** [comment]
I somehow can't reproduce the RtD error when running the notebook locally.

**Comment by RondeauG** [comment]
Alright, I managed to reproduce the error locally. It looks like the culprit is `xarray 2023.3.0`, which gets installed during:
`python -m pip install --upgrade --upgrade-strategy eager --no-cache-dir .`

Is there a way to prevent that? Incidentally, that command also installs `pandas 1.5.3`, even though we have `pandas!=1.5.3` in `environment.yml`. Is it because it's not in `setup.py`?

**Review by RondeauG — APPROVED** [review]
The docs seem to be generating correctly !

File: environment-dev.yml [hunk 1]
```diff
@@ -19,12 +20,12 @@ dependencies:
...
   - netCDF4
   - numpy
-  - pandas!=1.5.3
+  - pandas !=1.5.3
   - pygeos
   - pyyaml
   - rechunker
   - shapely
-  - xarray
+  - xarray <2023.3.0  # See https://github.com/Ouranosinc/xscen/pull/173
```
Comment (aulemahal) [inline environment-dev.yml:28]: I disagree with pinning a package like this, _especially_ xarray!

The bug is in xclim, and comes from a change in a semi-private function in xarray (issue 7645 there).

Comment (Zeitsperre) [reply environment-dev.yml:28]: Understood. How do we proceed? `xarray !=2023.3.0`?

Comment (Zeitsperre) [reply environment-dev.yml:28]: I imagine if we can push changes to `xclim`, we can simply pin `xclim` to the next version? Will changes in `xclim` be backwards-compatible with `xarray`?

Comment (RondeauG) [reply environment-dev.yml:28]: Done, I made an Issue.

Comment (Zeitsperre) [reply environment-dev.yml:28]: Awesome. I'll add the changes to the changelog and update the issue to consult, then merge. Thanks!

**Pull request merged**